tests/unit_tests/test_unit.py

```diff
@@ -139,6 +139,54 @@ def test_aa_pep_score():
     assert peptide_score == pytest.approx(-0.5)
 
 
+def test_calc_match_score():
+    """
+    Test the calculation of geometric scores using teacher-forced
+    decoder output probabilities and ground truth amino acid sequences.
+    """
+    first_slot_prob = torch.zeros(29)
+    first_slot_prob[1] = 1.0  # A
+    second_slot_prob = torch.zeros(29)
+    second_slot_prob[2] = 1.0  # B
+    third_slot_prob = torch.zeros(29)
+    third_slot_prob[3] = 1.0  # C
+    stop_slot_prob = torch.zeros(29)
+    stop_slot_prob[28] = 1.0  # $
+    blank_slot_prob = torch.zeros(29)
+
+    pep_1_aa = torch.stack(
+        [
+            first_slot_prob,
+            second_slot_prob,
+            third_slot_prob,
+            stop_slot_prob,
+            blank_slot_prob,
+        ]
+    )
+    pep_2_aa = torch.stack(
+        [
+            third_slot_prob,
+            second_slot_prob,
+            stop_slot_prob,
+            blank_slot_prob,
+            blank_slot_prob,
+        ]
+    )
+
+    batch_all_aa_scores = torch.stack([pep_1_aa, pep_2_aa])
+    truth_aa_indices = torch.tensor([[1, 2, 3, 28], [3, 2, 28, 0]])
+
+    all_scores, masked_per_aa_scores = _calc_match_score(
+        batch_all_aa_scores, truth_aa_indices
+    )
+
+    assert all_scores.numpy()[0] == pytest.approx(0)
```

Comment:
**todo:** We need to test a better range of scores, not only 0.

Reply:
Fixed, added additional tests